packages/electron-updater/src/BaseUpdater.ts

```diff
@@ -48,7 +48,20 @@ export abstract class BaseUpdater extends AppUpdater {
     }
 
     const downloadedUpdateHelper = this.downloadedUpdateHelper
-    const installerPath = downloadedUpdateHelper == null ? null : downloadedUpdateHelper.file
+    
+    // Get the installer path, ensuring spaces are escaped on Linux
+    // 1. Check if downloadedUpdateHelper is not null
+    // 2. Check if downloadedUpdateHelper.file is not null
+    // 3. If both checks pass:
+    //    a. If the platform is Linux, replace spaces with '\ ' for shell compatibility
+    //    b. If the platform is not Linux, use the original path
+    // 4. If any check fails, set installerPath to null
+    const installerPath = downloadedUpdateHelper && downloadedUpdateHelper.file 
+      ? (process.platform === 'linux' 
+        ? downloadedUpdateHelper.file.replace(/ /g, '\\ ') 
```

Comment:
Will this be an issue for AppImages? Would this break previous functionality with the `emit` logic?
https://github.com/electron-userland/electron-builder/blob/12c52a81420f04ec0e205dd83798c2b0b773011d/packages/electron-updater/src/AppImageUpdater.ts#L89-L96

Reply:
> Will this be an issue for AppImages? Would this break previous functionality with the `emit` logic?
> https://github.com/electron-userland/electron-builder/blob/12c52a81420f04ec0e205dd83798c2b0b773011d/packages/electron-updater/src/AppImageUpdater.ts#L89-L96

Thank you for pointing that! It looks like it shouldn't create any problems but in fact fix an error caused by the mv command if this was not implemented

Reply:
@mmaietta simulated this in node REPL.
![image](https://github.com/user-attachments/assets/be3a4f4c-f8e4-4b20-98ab-8b737f40b6b8)

Does not seem to create any problems and correctly `path.basename` recognized the escape sequence correctly and figured out `\ ` implies a space.